src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -47,15 +47,16 @@ public override async Task OnConnectedAsync(HubConnectionContext connection)
                 scope = _serviceScopeFactory.CreateScope();
 
                 var hubActivator = scope.ServiceProvider.GetRequiredService<IHubActivator<THub>>();
```

Comment:
Does it still need to be resolved per-request? (I haven't made this change in gRPC yet)

Is it because you've not change it yet, or are you thinking of people who might choose to make the activator scoped when they register there own type?

Reply:
No it doesn't but it doesn't matter. If it's registered as a singleton, it'll come from the right place.

Reply:
There is a little overhead for resolving the activator for a hub once each request vs once in the ctor though.

Reply:
True, I'll change it.

Reply:
Ugh, that changes a ton of public API...